clang/include/clang/Basic/Attr.td

```diff
@@ -1069,10 +1074,10 @@ def SYCLDevice : InheritableAttr {
 }
 
 def SYCLKernel : InheritableAttr {
-  let Spellings = [GNU<"sycl_kernel">];
-  let Subjects = SubjectList<[Function]>;
+  let Spellings = [Clang<"sycl_kernel">];
+  let Subjects = SubjectList<[FunctionTmpl]>;
   let LangOpts = [SYCLIsDevice];
```

Comment:
I think we should add SYCLIsHost too to suppress the warnings coming from host compilation.

Reply:
I disagree. This attribute is supposed be used by SYCL device compiler only! Host compiler must ignore this attribute and warn about this.

Reply:
In a perfect world host compiler shouldn't see this attribute at all in the code. But as SYCL is a single source language, so host compiler see the attribute anyway and warns users about it. I could live with that, but some people find it confusing.

Reply:
> host compiler see the attribute anyway

What? What code are you talking about?
All usages of this attribute is ifdef-ed with `__SYCL_DEVICE_ONLY__` check. Host compiler do not "see" the attribute in our implementation.